src/components/app/data/hooks/useEnterpriseCustomer.test.jsx

```diff
@@ -1,53 +1,110 @@
 import { queryClient } from '../../../../utils/tests';
-import { fetchEnterpriseLearnerData } from '../services';
+import { fetchEnterpriseLearnerDashboard, fetchEnterpriseLearnerData } from '../services';
 
 jest.mock('../services', () => ({
   ...jest.requireActual('../services'),
   fetchEnterpriseLearnerData: jest.fn().mockResolvedValue(null),
+  fetchEnterpriseLearnerDashboard: jest.fn().mockResolvedValue(null),
 }));
+
 const mockEnterpriseCustomer = enterpriseCustomerFactory();
 const mockAuthenticatedUser = authenticatedUserFactory();
 const mockEnterpriseLearnerData = {
   enterpriseCustomer: mockEnterpriseCustomer,
-  enterpriseCustomerUserRoleAssignments: [],
   activeEnterpriseCustomer: null,
   activeEnterpriseCustomerUserRoleAssignments: [],
   allLinkedEnterpriseCustomerUsers: [],
-  enterpriseFeatures: {},
   staffEnterpriseCustomer: null,
+  enterpriseFeatures: {
+    isBFFEnabled: false,
+  },
+  shouldUpdateActiveEnterpriseCustomerUser: false,
+};
+const mockBFFDashboardData = {
+  enterpriseCustomer: {
+    ...mockEnterpriseCustomer,
+    isBFFEnabled: true,
+  },
+  allLinkedEnterpriseCustomerUsers: [],
+  enterpriseFeatures: {
+    isBFFEnabled: true,
+  },
+  shouldUpdateActiveEnterpriseCustomerUser: false,
+  enterpriseCustomerUserSubsidies: {
+    subscriptions: {
+      customerAgreement: {},
+      subscriptionLicenses: [],
+      subscriptionLicensesByStatus: {
+        activated: [],
+        assigned: [],
+        expired: [],
+        revoked: [],
+      },
+    },
+  },
+  enterpriseCourseEnrollments: [],
+  errors: [],
+  warnings: [],
 };
 
+const mockExpectedEnterpriseCustomers = (isMatchedRoute) => (isMatchedRoute
+  ? mockBFFDashboardData.enterpriseCustomer
+  : mockEnterpriseLearnerData.enterpriseCustomer);
+
 describe('useEnterpriseCustomer', () => {
-  const Wrapper = ({ children }) => (
+  const Wrapper = ({ initialEntries = [], children }) => (
     <QueryClientProvider client={queryClient()}>
-      <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
-        {children}
-      </AppContext.Provider>
+      <MemoryRouter initialEntries={initialEntries}>
+        <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
+          <Routes>
+            <Route path=":enterpriseSlug" element={children} />
+            <Route path=":enterpriseSlug/search" element={children} />
+          </Routes>
+          {children}
+        </AppContext.Provider>
+      </MemoryRouter>
     </QueryClientProvider>
   );
   beforeEach(() => {
     jest.clearAllMocks();
     fetchEnterpriseLearnerData.mockResolvedValue(mockEnterpriseLearnerData);
+    fetchEnterpriseLearnerDashboard.mockResolvedValue(mockBFFDashboardData);
   });
-  it('should return enterprise customer metadata correctly', async () => {
-    const { result, waitForNextUpdate } = renderHook(() => useEnterpriseCustomer(), { wrapper: Wrapper });
+  it.each([
+    { isMatchedRoute: false },
```

Comment:
nit: perhaps `isMatchedBFFRoute` or `isBFFEnabledRoute` to clarify what "matched" means